src/lavinmq/shovel/shovel.cr

```diff
@@ -334,14 +351,60 @@ module LavinMQ
                else
                  "/"
                end
-        response = c.post(path, headers: headers, body: msg.body_io)
+        if @signature_secrets.empty?
+          # Stream body directly when signature is not needed
+          response = c.post(path, headers: headers, body: msg.body_io)
+        else
+          # Check payload size before reading into memory
+          if msg.body_io.bytesize > @max_signed_webhook_payload
+            Log.error { "Webhook payload size (#{msg.body_io.bytesize} bytes) exceeds max_signed_webhook_payload limit (#{@max_signed_webhook_payload} bytes), rejecting message without requeue" }
+            source.reject(msg.delivery_tag, requeue: false)
+            return
+          end
```

Comment:
The message is already in memory, `AMQP::Client::Message#body_io` is a `IO::Memory`. Calculating the HMAC over a 128MB message takes 100ms on my laptop. Uploading it with a 1Gbit/s connection takes 125ms. I'm not sure the complexity that this introduces (more config values, code etc) is worth it. 
```suggestion
```